pkg/fluxexec/install_test.go

```diff
@@ -0,0 +1,92 @@
+package fluxexec
+
+import (
+	"context"
+	"os/exec"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("installCmd", func() {
+	It("should be able to generate correct install commands", func() {
+		By("generating the default command", func() {
+			path, err := exec.LookPath("flux")
+			Expect(err).To(BeNil())
+
+			flux, err := NewFlux(".", path)
+			Expect(err).To(BeNil())
+
+			initCmd, err := flux.installCmd(context.TODO())
+			Expect(err).To(BeNil())
+			Expect(initCmd.Args[1:]).To(Equal([]string{
+				"install",
+				"--watch-all-namespaces",
+				"--cluster-domain",
+				"cluster.local",
+				"--log-level",
+				"info",
+				"--network-policy",
+				"--registry",
+				"ghcr.io/fluxcd",
+				"--components",
+				"source-controller,kustomize-controller,helm-controller,notification-controller",
+			}))
+		})
+		By("generating the command without network policy", func() {
+			path, err := exec.LookPath("flux")
+			Expect(err).To(BeNil())
+
+			flux, err := NewFlux(".", path)
+			Expect(err).To(BeNil())
+
+			initCmd, err := flux.installCmd(context.TODO(), NetworkPolicy(false))
+			Expect(err).To(BeNil())
+			Expect(initCmd.Args[1:]).To(Equal([]string{
+				"install",
+				"--watch-all-namespaces",
+				"--cluster-domain", "cluster.local",
+				"--log-level", "info",
+				"--registry", "ghcr.io/fluxcd",
+				"--components", "source-controller,kustomize-controller,helm-controller,notification-controller",
+			}))
+		})
+		By("generating the command to install only source controller", func() {
+			path, err := exec.LookPath("flux")
+			Expect(err).To(BeNil())
+
+			flux, err := NewFlux(".", path)
+			Expect(err).To(BeNil())
+
+			initCmd, err := flux.installCmd(context.TODO(), Components("source-controller"))
+			Expect(err).To(BeNil())
+			Expect(initCmd.Args[1:]).To(Equal([]string{
+				"install",
+				"--watch-all-namespaces",
+				"--cluster-domain", "cluster.local",
+				"--log-level", "info",
+				"--network-policy",
+				"--registry", "ghcr.io/fluxcd",
+				"--components", "source-controller",
+			}))
+		})
+		By("generating the command to install a wrong controller", func() {
+			path, err := exec.LookPath("flux")
+			Expect(err).To(BeNil())
+
+			flux, err := NewFlux(".", path)
+			Expect(err).To(BeNil())
+
+			initCmd, err := flux.installCmd(context.TODO(), Components("abc-controller"))
+			Expect(err).To(BeNil())
```

Comment:
Is this really good behaviour? Neither flux nor gitops will be able to report that it won't do what you (thought you) asked it to do. I always think image automation is enabled with "image-automation-controller" and this wouldn't give me any indication that it won't.

Reply:
The package should be able to filter out unsupported controllers. I'll double check this behavior.